Legal NLP tasks on Swiss data

rolshoven opened the pull request:
This PR introduces a  set of Swiss legal eval tasks, initially created by @JoelNiklaus and extended by me for the [SLDS dataset](https://huggingface.co/datasets/ipst/slds). I had to do some adaptions to the other included tasks as well due to recent changes in lighteval, but I think it should be working again now.

rolshoven committed:
Legal NLP tasks on Swiss data

NathanHB commented:
yes so what you could do is make a folder named after your task and split your file into separate modules.

- prompts.py: for the judge prompt and tasks prompts
- metrics.py: for your metric defs
- main.py: for your task defs

this should be easier to review and maintain!

rolshoven commented:
@NathanHB Sorry I was away for three weeks but now I'm back :) I hope that I finde some time soon to do the changes. But does that mean that the community task itself is a directory, which contains the three suggested files, and then only the `main.py` includes the `DATASETS` and `TASK_TABLE` variables?

rolshoven committed:
refactor: split Swiss legal multilingual tasks into modular package

Move Swiss legal task definitions into prompts/metrics/main modules and keep backward compatibility via swiss_legal_evals re-export.

rolshoven committed:
refactor: update higher_is_better type in MetricGrouping

Changed the type of higher_is_better from a dictionary of callables to a dictionary of booleans for improved clarity and type safety. This is also how it has been used so far.

rolshoven committed:
refactor: Updated prompts and implementation to match the latest SwiLTra-Bench code

rolshoven committed:
refactor: Enhance COMET and GEMBA metric loading with error handling

- Introduced functions to load COMET and GEMBA metrics, providing clear error messages for missing dependencies.
- Disabled specific COMET metrics due to numpy version conflicts, with warnings logged for skipped metrics.
- Updated GPU metrics list to reflect the disabled COMET metrics.
- Improved code organization and clarity in metric processing functions.

rolshoven committed:
Add Gemba dependency for Swiss legal evaluations and remove `suite` parameter from TranslationTask constructor.

rolshoven committed:
Fix batched metric aggregation for grouped metric names

rolshoven committed:
Fixed missing system prompt

rolshoven committed:
Judge models now are used through OpenRouter

rolshoven committed:
fix reasoning model token handling when max_tokens is unset

rolshoven commented:
@NathanHB A little bit late, but I implemented the proposed changes and fixed some bugs. Could you revise the changes and give feedback? Thank you :)

rolshoven committed:
chore: trigger PR update

rolshoven commented:
@NathanHB Friendly ping -- just wanted to check if you had a chance to look at this 🙂

NathanHB committed:
Merge branch 'main' into community_task_slds

NathanHB commented on src/lighteval/tasks/multilingual/tasks/swiss_legal/metrics.py:
```diff
@@ -0,0 +1,934 @@
+        prediction = model_response.final_text[0]  # Get first prediction
+
+        score = sentence_bleu(prediction, [gold]).score
+        return score * 100
```
isn't bleu score already between (0 - 100) ?

rolshoven replied:
You're right, I fixed that by removing the scaling from BLEU and also from CHRF and TER, thanks for pointing this out!

NathanHB commented on src/lighteval/tasks/multilingual/tasks/swiss_legal/metrics.py:
```diff
@@ -0,0 +1,934 @@
+
+class BertScoreMultilingual(BertScore):
+    def __init__(
+        self, normalize_gold=None, normalize_pred=None, language=str, model_type=str, num_layers=int, device=str
```
```suggestion
        self, normalize_gold=None, normalize_pred=None, language:str, model_type:str, num_layers:int, device:str
```

Type annotation uses ":"

rolshoven replied:
Oh wow no idea how that got there, I'm glad you noticed! All arguments in `__init__` now have type hints and the default values are set.

NathanHB commented on src/lighteval/tasks/multilingual/tasks/swiss_legal/metrics.py:
```diff
@@ -0,0 +1,934 @@
+        elif self.language == "fr":
+            one_shot_example = SLDS_JUDGE_ONE_SHOT_EXAMPLE_FR.strip()
+        elif self.language == "it":
+            one_shot_example = SLDS_JUDGE_ONE_SHOT_EXAMPLE_IT.strip()
```
Add an else clause, either trhowing an error or setting the `one_shot_example` var

rolshoven replied:
Done!

NathanHB reviewed (commented):
Sorry for the delay on this! Approving the PR so you can merge asap, only few nits to fix / be aware of.

rolshoven committed:
fix: return raw score for BLEU, CHRF, and TER metrics instead of scaled values

rolshoven committed:
fix: replaced accidental default value assignment with intended type hint

rolshoven committed:
fix: add error handling for unsupported languages in Swiss Landmark Decision Summarization judge

rolshoven committed:
fix: avoid huge negative BERTScore from baseline rescaling

Default `rescale_with_baseline` to False in BertScoreMultilingual. With
near-1.0 baselines (e.g. German xlm-roberta-large layer-24 ≈ 0.98), the
(score - baseline) / (1 - baseline) formula amplifies deviations ~50x,
and the subsequent x100 scaling compounds it — empty/weak predictions
could yield scores like -5000.
Make rescaling opt-in, warn when enabled, and skip the x100 scaling in
that case since rescaled scores are not bounded to [0, 1].

rolshoven commented:
> Sorry for the delay on this! Approving the PR so you can merge asap, only few nits to fix / be aware of.

@NathanHB Thank you for having a look! I made the suggested changes and I also set `rescale_with_baseline` to False, since it may lead to large negative scores for specific outputs (such as empty responses):

https://github.com/rolshoven/lighteval/blob/df00dc169970849cced0a8ae2be88fbc5f277297/src/lighteval/tasks/multilingual/tasks/swiss_legal/metrics.py#L102-L110

If there is anything else that you think should change, let me know! Otherwise I'm looking forward to having the changes integrated :-)
